includes/classes/InternalConnections/NetworkSiteConnection.php

```diff
@@ -890,22 +889,29 @@ public static function build_available_authorized_sites( $user_id = false, $cont
 	 * will have caching enabled, so we also store it
 	 * in a site option.
 	 *
-	 * @return void
+	 * @return string
 	 */
 	public static function set_sites_last_changed_time() {
-		update_site_option( 'last_changed_sites', microtime() );
+		$time = microtime();
+		update_site_option( 'last_changed_sites', $time );
+
+		return $time;
 	}
 
 	/**
-	 * Rebuild the authorized sites cache for a specific user.
+	 * Clear the authorized sites cache for a specific user.
 	 *
 	 * @param int $user_id Current user ID.
-	 *
-	 * @return void
 	 */
-	public static function rebuild_user_authorized_sites_cache( $user_id ) {
-		self::build_available_authorized_sites( $user_id, 'push', true );
-		self::build_available_authorized_sites( $user_id, 'pull', true );
+	public static function clear_authorized_sites_cache( $user_id = false ) {
+		$last_changed = get_site_option( 'last_changed_sites' );
+
+		if ( ! $last_changed ) {
+			$last_changed = self::set_sites_last_changed_time();
```

Comment:
@dkotter In case of just-set `$last_changed`, both `delete_transient` functions will address non-existing transients, are we sure this is ok?

Reply:
@cadic I don't think it causes any issues to try deleting a transient that doesn't exist but I've modified this slightly so we only delete these transients if we have a proper `last_changed_sites` value. Otherwise the transients should either not exist (so they don't need deleted) or they exist with a different `last_changed_sites` value and will now be invalid since we've updated that value